build: Bump libevent minimum version up to 2.1.8

hebasto opened the pull request:
Required to support new functionality in bitcoin/bitcoin#19420.

`libevent` availability: https://repology.org/project/libevent/versions

hebasto committed:
build: Bump libevent minimum version up to 2.1.8

Required to support new functionality.

hebasto commented:
cc @promag

promag reviewed (commented):
ACK aaf72d62c18f9cb325c150cf0cc21abb201607c8 🙏

prusnak commented:
For reference:
* 2.0.21 was released in 2012
* 2.1.8 was released in 2017

fanquake committed:
refactor: account for requiring libevent 2.1.8+

fanquake commented:
I think requiring 2.1.8 from 24.x onwards is ok, if there is something we want to use, but shouldn't this PR have more changes? The version number for the 2.1.8 release is `0x02010800`, so all our `#if LIBEVENT_VERSION_NUMBER >= 0x02010300` conditions should go, as well as a workaround in the fuzz tests. See branch here: https://github.com/fanquake/bitcoin/tree/account_for_new_libevent_minimum.

hebasto commented:
@fanquake 

> I think requiring 2.1.8 from 24.x onwards is ok, if there is something we want to use, but shouldn't this PR have more changes? The version number for the 2.1.8 release is `0x02010800`, so all our `#if LIBEVENT_VERSION_NUMBER >= 0x02010300` conditions should go, as well as a workaround in the fuzz tests. See branch here: https://github.com/fanquake/bitcoin/tree/account_for_new_libevent_minimum.

Thanks! Added.

MarcoFalke commented on src/bitcoin-cli.cpp:
```diff
@@ -747,9 +743,8 @@ static UniValue CallRPC(BaseRequestHandler* rh, const std::string& strMethod, co
     if (req == nullptr)
         throw std::runtime_error("create http request failed");
-#if LIBEVENT_VERSION_NUMBER >= 0x02010300
+
```
```suggestion
}
```

nit: Maybe add `{}` while touching the line?

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/24681#issuecomment-1083053772).

hebasto commented:
Updated a362c43875b0974c23238e55bc3f21c224fe84c0 -> 0598f36852199d0cee8fe9e676a2e0bec3ebf624 ([pr24681.03](https://github.com/hebasto/bitcoin/commits/pr24681.03) -> [pr24681.04](https://github.com/hebasto/bitcoin/commits/pr24681.04), [diff](https://github.com/hebasto/bitcoin/compare/pr24681.03..pr24681.04)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/24681#discussion_r838453030)

hebasto commented:
#### Guix builds on `x86_64`:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
b7738a0c94aaaa58a2bc8846b60c37c565184451c92dc6b856c0dc9d5e0d4197  guix-build-0598f3685219/output/aarch64-linux-gnu/SHA256SUMS.part
4808ec36d368cf65846e4de6e7e206f5d262e55522511bf80c6267f4011a8f4c  guix-build-0598f3685219/output/aarch64-linux-gnu/bitcoin-0598f3685219-aarch64-linux-gnu-debug.tar.gz
6897a48e2289581f065673730a09fdd6697a00542f340cd965a39755c07b7549  guix-build-0598f3685219/output/aarch64-linux-gnu/bitcoin-0598f3685219-aarch64-linux-gnu.tar.gz
df617c76e0bc96e3e543ccc374bcdf7e080037fd48960e9569c5190a33d14499  guix-build-0598f3685219/output/arm-linux-gnueabihf/SHA256SUMS.part
4292972310cb72b4ed7a22e631577826ff46bf23afbebbed17e1ed8d23a7d20e  guix-build-0598f3685219/output/arm-linux-gnueabihf/bitcoin-0598f3685219-arm-linux-gnueabihf-debug.tar.gz
f346d3acd2bdcb76e6e1f59a98d71ddc5827b6efa78527b23955f1e75ac3df5d  guix-build-0598f3685219/output/arm-linux-gnueabihf/bitcoin-0598f3685219-arm-linux-gnueabihf.tar.gz
3690a84b8272491a0b455695c4e6bd7223d6f8f499db910c8476f35ed07724c2  guix-build-0598f3685219/output/arm64-apple-darwin/SHA256SUMS.part
363d343c74e3a60d18c8fc22fc7b52a6af9157c9f6f106654be3fa3036365930  guix-build-0598f3685219/output/arm64-apple-darwin/bitcoin-0598f3685219-arm64-apple-darwin-unsigned.dmg
6491aec3025c45ee95a917062e1e044e128754ca420ed5dfb7de37558a7b7499  guix-build-0598f3685219/output/arm64-apple-darwin/bitcoin-0598f3685219-arm64-apple-darwin-unsigned.tar.gz
eca770ebc37ca081b2a88247738a001df16a34e61f2e997c268b306625bd38b9  guix-build-0598f3685219/output/arm64-apple-darwin/bitcoin-0598f3685219-arm64-apple-darwin.tar.gz
9ebb415c90efc517a603e68131ff06f9c7442e95e0d2e3b08f65523792922f42  guix-build-0598f3685219/output/dist-archive/bitcoin-0598f3685219.tar.gz
8d62b2932975f864348f113c482ce19ebf601c03c12d1952a3b53ff163f43203  guix-build-0598f3685219/output/powerpc64-linux-gnu/SHA256SUMS.part
a897b71ab60bf0150620ea90da601ebd9414305e7789ebfe68020e58e3642166  guix-build-0598f3685219/output/powerpc64-linux-gnu/bitcoin-0598f3685219-powerpc64-linux-gnu-debug.tar.gz
d9f9506286b5ef3dceee31d9742c24b89f1140e92e1a8df98da381b64c2d767b  guix-build-0598f3685219/output/powerpc64-linux-gnu/bitcoin-0598f3685219-powerpc64-linux-gnu.tar.gz
924fa61ff89895d0711222056907617b23a761e0ae56063423dacd0a3de20cce  guix-build-0598f3685219/output/powerpc64le-linux-gnu/SHA256SUMS.part
76a0719f342f44741f99435b3ff5576af74b5d55c39a1af64dbd536f56ab52f6  guix-build-0598f3685219/output/powerpc64le-linux-gnu/bitcoin-0598f3685219-powerpc64le-linux-gnu-debug.tar.gz
8423d01ac8efa523047b0b895f55e0efda6289666ad7526b92ade2c192605949  guix-build-0598f3685219/output/powerpc64le-linux-gnu/bitcoin-0598f3685219-powerpc64le-linux-gnu.tar.gz
2c9848faecff7e9f9c711f9ac32fa6deceee401a883b72a11d117b4f8d1d34f7  guix-build-0598f3685219/output/riscv64-linux-gnu/SHA256SUMS.part
1060ae38588ca98782a6fe7e0763e565fd1a115c7dc6f42b97be95f4039e48a3  guix-build-0598f3685219/output/riscv64-linux-gnu/bitcoin-0598f3685219-riscv64-linux-gnu-debug.tar.gz
f7401fdc281e85b9ee2ae9f578d03aba0ee641aebdf18fcebf44cac88ffb0a52  guix-build-0598f3685219/output/riscv64-linux-gnu/bitcoin-0598f3685219-riscv64-linux-gnu.tar.gz
d46b89599cea6b2959de8ea2e9dc5a37ad0ac973b837c95acba879c79820a9d6  guix-build-0598f3685219/output/x86_64-apple-darwin/SHA256SUMS.part
564a392ca2f66f354d2d9c934b0539e9b787eb1f8c8679b9ab9a9281532ff1e0  guix-build-0598f3685219/output/x86_64-apple-darwin/bitcoin-0598f3685219-x86_64-apple-darwin-unsigned.dmg
94bdb7b18764d0a9b6898e977e6a93b2fef9e435e08c3186b4dedb21780b460a  guix-build-0598f3685219/output/x86_64-apple-darwin/bitcoin-0598f3685219-x86_64-apple-darwin-unsigned.tar.gz
83aad0ef1701537bf40f75d5440f1757b3d2234850f095c24f902bacedc6422e  guix-build-0598f3685219/output/x86_64-apple-darwin/bitcoin-0598f3685219-x86_64-apple-darwin.tar.gz
757a62da0e2f658eecc820c11007e73323ce3c7c1bc1128b42204a763a0028bd  guix-build-0598f3685219/output/x86_64-linux-gnu/SHA256SUMS.part
8cd9dedf07a2a5f0f0e8f28f3786e417bc8d1c260ef02ce9734fc3c27a1d34fb  guix-build-0598f3685219/output/x86_64-linux-gnu/bitcoin-0598f3685219-x86_64-linux-gnu-debug.tar.gz
dbdc614531ad45ba76b87fe9ce22c0365241f29e5b24d4fcab5e80194199a7df  guix-build-0598f3685219/output/x86_64-linux-gnu/bitcoin-0598f3685219-x86_64-linux-gnu.tar.gz
251bd427c27a94e34159eb133425e1193ddfa07bdfa372947d2d428b7331cdad  guix-build-0598f3685219/output/x86_64-w64-mingw32/SHA256SUMS.part
6f8cd99f21f7d6f327532d8a6ece209e3182691b48c2fa777b0f7ef24051906d  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64-debug.zip
4d1a88a15cea2a3e227f6a6b6add3ce1c49a4bbfcf20c2e2df4e94efa3b43084  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64-setup-unsigned.exe
6d53f8afb56ee66ca97ac8137ab571ced604cb765bf19170278fd1f680187b46  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64-unsigned.tar.gz
93f44aa2fa255f577bc13f2cc008f73d15ba2cee0e9797d3cb8084008107ef76  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64.zip
```

jonatack reviewed (commented):
Concept ACK per https://github.com/bitcoin/bitcoin/pull/19420#issuecomment-1076258084.

utACK 0598f36852199d0cee8fe9e676a2e0bec3ebf624

Light-GitHub-and-grep-locally-review while waiting on a build of another pull; it looks like all the instances of `LIBEVENT_VERSION_NUMBER`, `0x02010100` and `= 2.0.21` have been handled

laanwj commented:
A reason to not bump the minimum libevent from 2.0.x was that distributions didn't ship libevent 2.1.x versions (due to some unclarity about beta versus stable). Did this get resolved, do distros now reliably ship 2.1.8? or above?

laanwj commented:
Concept ACK then.

prusnak reviewed (approved):
Concept ACK

laanwj commented on src/test/fuzz/http_request.cpp:
```diff
@@ -19,23 +19,8 @@
 #include <string>
 #include <vector>
 
-// workaround for libevent versions before 2.1.1,
```
Nice to get rid of cruft like this.

fanquake commented:
Guix Build (on x86_64):
```bash
b7738a0c94aaaa58a2bc8846b60c37c565184451c92dc6b856c0dc9d5e0d4197  guix-build-0598f3685219/output/aarch64-linux-gnu/SHA256SUMS.part
4808ec36d368cf65846e4de6e7e206f5d262e55522511bf80c6267f4011a8f4c  guix-build-0598f3685219/output/aarch64-linux-gnu/bitcoin-0598f3685219-aarch64-linux-gnu-debug.tar.gz
6897a48e2289581f065673730a09fdd6697a00542f340cd965a39755c07b7549  guix-build-0598f3685219/output/aarch64-linux-gnu/bitcoin-0598f3685219-aarch64-linux-gnu.tar.gz
df617c76e0bc96e3e543ccc374bcdf7e080037fd48960e9569c5190a33d14499  guix-build-0598f3685219/output/arm-linux-gnueabihf/SHA256SUMS.part
4292972310cb72b4ed7a22e631577826ff46bf23afbebbed17e1ed8d23a7d20e  guix-build-0598f3685219/output/arm-linux-gnueabihf/bitcoin-0598f3685219-arm-linux-gnueabihf-debug.tar.gz
f346d3acd2bdcb76e6e1f59a98d71ddc5827b6efa78527b23955f1e75ac3df5d  guix-build-0598f3685219/output/arm-linux-gnueabihf/bitcoin-0598f3685219-arm-linux-gnueabihf.tar.gz
3690a84b8272491a0b455695c4e6bd7223d6f8f499db910c8476f35ed07724c2  guix-build-0598f3685219/output/arm64-apple-darwin/SHA256SUMS.part
363d343c74e3a60d18c8fc22fc7b52a6af9157c9f6f106654be3fa3036365930  guix-build-0598f3685219/output/arm64-apple-darwin/bitcoin-0598f3685219-arm64-apple-darwin-unsigned.dmg
6491aec3025c45ee95a917062e1e044e128754ca420ed5dfb7de37558a7b7499  guix-build-0598f3685219/output/arm64-apple-darwin/bitcoin-0598f3685219-arm64-apple-darwin-unsigned.tar.gz
eca770ebc37ca081b2a88247738a001df16a34e61f2e997c268b306625bd38b9  guix-build-0598f3685219/output/arm64-apple-darwin/bitcoin-0598f3685219-arm64-apple-darwin.tar.gz
9ebb415c90efc517a603e68131ff06f9c7442e95e0d2e3b08f65523792922f42  guix-build-0598f3685219/output/dist-archive/bitcoin-0598f3685219.tar.gz
8d62b2932975f864348f113c482ce19ebf601c03c12d1952a3b53ff163f43203  guix-build-0598f3685219/output/powerpc64-linux-gnu/SHA256SUMS.part
a897b71ab60bf0150620ea90da601ebd9414305e7789ebfe68020e58e3642166  guix-build-0598f3685219/output/powerpc64-linux-gnu/bitcoin-0598f3685219-powerpc64-linux-gnu-debug.tar.gz
d9f9506286b5ef3dceee31d9742c24b89f1140e92e1a8df98da381b64c2d767b  guix-build-0598f3685219/output/powerpc64-linux-gnu/bitcoin-0598f3685219-powerpc64-linux-gnu.tar.gz
924fa61ff89895d0711222056907617b23a761e0ae56063423dacd0a3de20cce  guix-build-0598f3685219/output/powerpc64le-linux-gnu/SHA256SUMS.part
76a0719f342f44741f99435b3ff5576af74b5d55c39a1af64dbd536f56ab52f6  guix-build-0598f3685219/output/powerpc64le-linux-gnu/bitcoin-0598f3685219-powerpc64le-linux-gnu-debug.tar.gz
8423d01ac8efa523047b0b895f55e0efda6289666ad7526b92ade2c192605949  guix-build-0598f3685219/output/powerpc64le-linux-gnu/bitcoin-0598f3685219-powerpc64le-linux-gnu.tar.gz
2c9848faecff7e9f9c711f9ac32fa6deceee401a883b72a11d117b4f8d1d34f7  guix-build-0598f3685219/output/riscv64-linux-gnu/SHA256SUMS.part
1060ae38588ca98782a6fe7e0763e565fd1a115c7dc6f42b97be95f4039e48a3  guix-build-0598f3685219/output/riscv64-linux-gnu/bitcoin-0598f3685219-riscv64-linux-gnu-debug.tar.gz
f7401fdc281e85b9ee2ae9f578d03aba0ee641aebdf18fcebf44cac88ffb0a52  guix-build-0598f3685219/output/riscv64-linux-gnu/bitcoin-0598f3685219-riscv64-linux-gnu.tar.gz
d46b89599cea6b2959de8ea2e9dc5a37ad0ac973b837c95acba879c79820a9d6  guix-build-0598f3685219/output/x86_64-apple-darwin/SHA256SUMS.part
564a392ca2f66f354d2d9c934b0539e9b787eb1f8c8679b9ab9a9281532ff1e0  guix-build-0598f3685219/output/x86_64-apple-darwin/bitcoin-0598f3685219-x86_64-apple-darwin-unsigned.dmg
94bdb7b18764d0a9b6898e977e6a93b2fef9e435e08c3186b4dedb21780b460a  guix-build-0598f3685219/output/x86_64-apple-darwin/bitcoin-0598f3685219-x86_64-apple-darwin-unsigned.tar.gz
83aad0ef1701537bf40f75d5440f1757b3d2234850f095c24f902bacedc6422e  guix-build-0598f3685219/output/x86_64-apple-darwin/bitcoin-0598f3685219-x86_64-apple-darwin.tar.gz
757a62da0e2f658eecc820c11007e73323ce3c7c1bc1128b42204a763a0028bd  guix-build-0598f3685219/output/x86_64-linux-gnu/SHA256SUMS.part
8cd9dedf07a2a5f0f0e8f28f3786e417bc8d1c260ef02ce9734fc3c27a1d34fb  guix-build-0598f3685219/output/x86_64-linux-gnu/bitcoin-0598f3685219-x86_64-linux-gnu-debug.tar.gz
dbdc614531ad45ba76b87fe9ce22c0365241f29e5b24d4fcab5e80194199a7df  guix-build-0598f3685219/output/x86_64-linux-gnu/bitcoin-0598f3685219-x86_64-linux-gnu.tar.gz
251bd427c27a94e34159eb133425e1193ddfa07bdfa372947d2d428b7331cdad  guix-build-0598f3685219/output/x86_64-w64-mingw32/SHA256SUMS.part
6f8cd99f21f7d6f327532d8a6ece209e3182691b48c2fa777b0f7ef24051906d  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64-debug.zip
4d1a88a15cea2a3e227f6a6b6add3ce1c49a4bbfcf20c2e2df4e94efa3b43084  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64-setup-unsigned.exe
6d53f8afb56ee66ca97ac8137ab571ced604cb765bf19170278fd1f680187b46  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64-unsigned.tar.gz
93f44aa2fa255f577bc13f2cc008f73d15ba2cee0e9797d3cb8084008107ef76  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64.zip
```

fjahr committed:
refactor: Remove outdated libevent logging code

The removed code was intended to catch issues with event_enable_debug_logging which was not available prior to libevent 2.1.1. This is not necessary since the minimum libevent version was bumped to 2.1.8.

fjahr commented:
Concept ACK and light code review ACK 0598f36852199d0cee8fe9e676a2e0bec3ebf624

I think I have found another piece of code that can be refactored after the bump: https://github.com/fjahr/bitcoin/commit/e40779a4fee03c6c455149bd8e9d1a7ccd991450

If this is as straight forward as it seems to me, feel free to pull the commit in here. Otherwise I can open a follow-up PR.

prusnak commented:
> I think I have found another piece of code that can be refactored after the bump: [fjahr@e40779a](https://github.com/fjahr/bitcoin/commit/e40779a4fee03c6c455149bd8e9d1a7ccd991450)

+1 for including this change into the PR

jarolrod commented:
GUIX hashes on x86, mine match fanquake's and hebasto's latest round;

```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
b7738a0c94aaaa58a2bc8846b60c37c565184451c92dc6b856c0dc9d5e0d4197  guix-build-0598f3685219/output/aarch64-linux-gnu/SHA256SUMS.part
4808ec36d368cf65846e4de6e7e206f5d262e55522511bf80c6267f4011a8f4c  guix-build-0598f3685219/output/aarch64-linux-gnu/bitcoin-0598f3685219-aarch64-linux-gnu-debug.tar.gz
6897a48e2289581f065673730a09fdd6697a00542f340cd965a39755c07b7549  guix-build-0598f3685219/output/aarch64-linux-gnu/bitcoin-0598f3685219-aarch64-linux-gnu.tar.gz
df617c76e0bc96e3e543ccc374bcdf7e080037fd48960e9569c5190a33d14499  guix-build-0598f3685219/output/arm-linux-gnueabihf/SHA256SUMS.part
4292972310cb72b4ed7a22e631577826ff46bf23afbebbed17e1ed8d23a7d20e  guix-build-0598f3685219/output/arm-linux-gnueabihf/bitcoin-0598f3685219-arm-linux-gnueabihf-debug.tar.gz
f346d3acd2bdcb76e6e1f59a98d71ddc5827b6efa78527b23955f1e75ac3df5d  guix-build-0598f3685219/output/arm-linux-gnueabihf/bitcoin-0598f3685219-arm-linux-gnueabihf.tar.gz
3690a84b8272491a0b455695c4e6bd7223d6f8f499db910c8476f35ed07724c2  guix-build-0598f3685219/output/arm64-apple-darwin/SHA256SUMS.part
363d343c74e3a60d18c8fc22fc7b52a6af9157c9f6f106654be3fa3036365930  guix-build-0598f3685219/output/arm64-apple-darwin/bitcoin-0598f3685219-arm64-apple-darwin-unsigned.dmg
6491aec3025c45ee95a917062e1e044e128754ca420ed5dfb7de37558a7b7499  guix-build-0598f3685219/output/arm64-apple-darwin/bitcoin-0598f3685219-arm64-apple-darwin-unsigned.tar.gz
eca770ebc37ca081b2a88247738a001df16a34e61f2e997c268b306625bd38b9  guix-build-0598f3685219/output/arm64-apple-darwin/bitcoin-0598f3685219-arm64-apple-darwin.tar.gz
9ebb415c90efc517a603e68131ff06f9c7442e95e0d2e3b08f65523792922f42  guix-build-0598f3685219/output/dist-archive/bitcoin-0598f3685219.tar.gz
8d62b2932975f864348f113c482ce19ebf601c03c12d1952a3b53ff163f43203  guix-build-0598f3685219/output/powerpc64-linux-gnu/SHA256SUMS.part
a897b71ab60bf0150620ea90da601ebd9414305e7789ebfe68020e58e3642166  guix-build-0598f3685219/output/powerpc64-linux-gnu/bitcoin-0598f3685219-powerpc64-linux-gnu-debug.tar.gz
d9f9506286b5ef3dceee31d9742c24b89f1140e92e1a8df98da381b64c2d767b  guix-build-0598f3685219/output/powerpc64-linux-gnu/bitcoin-0598f3685219-powerpc64-linux-gnu.tar.gz
924fa61ff89895d0711222056907617b23a761e0ae56063423dacd0a3de20cce  guix-build-0598f3685219/output/powerpc64le-linux-gnu/SHA256SUMS.part
76a0719f342f44741f99435b3ff5576af74b5d55c39a1af64dbd536f56ab52f6  guix-build-0598f3685219/output/powerpc64le-linux-gnu/bitcoin-0598f3685219-powerpc64le-linux-gnu-debug.tar.gz
8423d01ac8efa523047b0b895f55e0efda6289666ad7526b92ade2c192605949  guix-build-0598f3685219/output/powerpc64le-linux-gnu/bitcoin-0598f3685219-powerpc64le-linux-gnu.tar.gz
2c9848faecff7e9f9c711f9ac32fa6deceee401a883b72a11d117b4f8d1d34f7  guix-build-0598f3685219/output/riscv64-linux-gnu/SHA256SUMS.part
1060ae38588ca98782a6fe7e0763e565fd1a115c7dc6f42b97be95f4039e48a3  guix-build-0598f3685219/output/riscv64-linux-gnu/bitcoin-0598f3685219-riscv64-linux-gnu-debug.tar.gz
f7401fdc281e85b9ee2ae9f578d03aba0ee641aebdf18fcebf44cac88ffb0a52  guix-build-0598f3685219/output/riscv64-linux-gnu/bitcoin-0598f3685219-riscv64-linux-gnu.tar.gz
d46b89599cea6b2959de8ea2e9dc5a37ad0ac973b837c95acba879c79820a9d6  guix-build-0598f3685219/output/x86_64-apple-darwin/SHA256SUMS.part
564a392ca2f66f354d2d9c934b0539e9b787eb1f8c8679b9ab9a9281532ff1e0  guix-build-0598f3685219/output/x86_64-apple-darwin/bitcoin-0598f3685219-x86_64-apple-darwin-unsigned.dmg
94bdb7b18764d0a9b6898e977e6a93b2fef9e435e08c3186b4dedb21780b460a  guix-build-0598f3685219/output/x86_64-apple-darwin/bitcoin-0598f3685219-x86_64-apple-darwin-unsigned.tar.gz
83aad0ef1701537bf40f75d5440f1757b3d2234850f095c24f902bacedc6422e  guix-build-0598f3685219/output/x86_64-apple-darwin/bitcoin-0598f3685219-x86_64-apple-darwin.tar.gz
757a62da0e2f658eecc820c11007e73323ce3c7c1bc1128b42204a763a0028bd  guix-build-0598f3685219/output/x86_64-linux-gnu/SHA256SUMS.part
8cd9dedf07a2a5f0f0e8f28f3786e417bc8d1c260ef02ce9734fc3c27a1d34fb  guix-build-0598f3685219/output/x86_64-linux-gnu/bitcoin-0598f3685219-x86_64-linux-gnu-debug.tar.gz
dbdc614531ad45ba76b87fe9ce22c0365241f29e5b24d4fcab5e80194199a7df  guix-build-0598f3685219/output/x86_64-linux-gnu/bitcoin-0598f3685219-x86_64-linux-gnu.tar.gz
251bd427c27a94e34159eb133425e1193ddfa07bdfa372947d2d428b7331cdad  guix-build-0598f3685219/output/x86_64-w64-mingw32/SHA256SUMS.part
6f8cd99f21f7d6f327532d8a6ece209e3182691b48c2fa777b0f7ef24051906d  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64-debug.zip
4d1a88a15cea2a3e227f6a6b6add3ce1c49a4bbfcf20c2e2df4e94efa3b43084  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64-setup-unsigned.exe
6d53f8afb56ee66ca97ac8137ab571ced604cb765bf19170278fd1f680187b46  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64-unsigned.tar.gz
93f44aa2fa255f577bc13f2cc008f73d15ba2cee0e9797d3cb8084008107ef76  guix-build-0598f3685219/output/x86_64-w64-mingw32/bitcoin-0598f3685219-win64.zip
```

hebasto commented:
@fjahr 

> Concept ACK and light code review ACK [0598f36](https://github.com/bitcoin/bitcoin/commit/0598f36852199d0cee8fe9e676a2e0bec3ebf624)
> 
> I think I have found another piece of code that can be refactored after the bump: [fjahr@e40779a](https://github.com/fjahr/bitcoin/commit/e40779a4fee03c6c455149bd8e9d1a7ccd991450)
> 
> If this is as straight forward as it seems to me, feel free to pull the commit in here. Otherwise I can open a follow-up PR.

Thanks! Added.

laanwj commented:
Code review ACK e40779a4fee03c6c455149bd8e9d1a7ccd991450

fanquake commented:
Guix Build (on x86_64):
```bash
61afdfbd14469eda67b2d915c1d9d387aa7cfa67c852f8b447f2a4c2ecefa64c  guix-build-e40779a4fee0/output/aarch64-linux-gnu/SHA256SUMS.part
43976291608acfee150772329ef583dcca9bfc3f2d89b8b1f8ff770663472c35  guix-build-e40779a4fee0/output/aarch64-linux-gnu/bitcoin-e40779a4fee0-aarch64-linux-gnu-debug.tar.gz
cde2ab4b3655bdc35a3828547c202f356319d107270d10b28c37e2772d290c86  guix-build-e40779a4fee0/output/aarch64-linux-gnu/bitcoin-e40779a4fee0-aarch64-linux-gnu.tar.gz
0eb72862a7f680c4274b9ff28dcd119206ff4a6c342d932cb606d3d9abb4da9c  guix-build-e40779a4fee0/output/arm-linux-gnueabihf/SHA256SUMS.part
fe122ba9e66d64a750f334f422bb74eea05dedc095cfb51558a623f60fb1d1c6  guix-build-e40779a4fee0/output/arm-linux-gnueabihf/bitcoin-e40779a4fee0-arm-linux-gnueabihf-debug.tar.gz
b413a3a3fa9e6e511332fa1ad25d53be4aa7a050edc485ecb055c975cbf5f60d  guix-build-e40779a4fee0/output/arm-linux-gnueabihf/bitcoin-e40779a4fee0-arm-linux-gnueabihf.tar.gz
decbd666758482433a0ded0c0e57552a47c33086096bb13c7877b000a83d07fd  guix-build-e40779a4fee0/output/arm64-apple-darwin/SHA256SUMS.part
bc041375b82629b4310238648080dfdcd6826b1631ddc6e7d4ca3243fabf42a4  guix-build-e40779a4fee0/output/arm64-apple-darwin/bitcoin-e40779a4fee0-arm64-apple-darwin-unsigned.dmg
ee5d1de4083488c00817f6f4b4a90b8ccc74b3caf29af0a3db49e04c00cf13d2  guix-build-e40779a4fee0/output/arm64-apple-darwin/bitcoin-e40779a4fee0-arm64-apple-darwin-unsigned.tar.gz
fcef23e20cc25dd99e9634875305a9a857a853d0a3bc3a88c51f5abb38241019  guix-build-e40779a4fee0/output/arm64-apple-darwin/bitcoin-e40779a4fee0-arm64-apple-darwin.tar.gz
22fad677c2e5f5610200f6fcfffe9bf61b1ca839799a34758832fc864d79459c  guix-build-e40779a4fee0/output/dist-archive/bitcoin-e40779a4fee0.tar.gz
f571e2c6b902c053289f77b06526f6ff84954e7054cf882695dbd9201c526142  guix-build-e40779a4fee0/output/powerpc64-linux-gnu/SHA256SUMS.part
4df2cb2fa948627e78588a2e7ed9f260c04f56cdebcbd880c0f70e014c2cdb58  guix-build-e40779a4fee0/output/powerpc64-linux-gnu/bitcoin-e40779a4fee0-powerpc64-linux-gnu-debug.tar.gz
827e8f11b56ede4d236ad17d7031538cd80caa9223276058cbc6614061edcd00  guix-build-e40779a4fee0/output/powerpc64-linux-gnu/bitcoin-e40779a4fee0-powerpc64-linux-gnu.tar.gz
b1fbf0768494bd4167a5bd43ed8c6ca08a089ba8a1b29d164bc517e2c662ce6b  guix-build-e40779a4fee0/output/powerpc64le-linux-gnu/SHA256SUMS.part
b5d7a26d860749f300f36a22471c4226ab9ec140a3f192e456855b6d78e34b4b  guix-build-e40779a4fee0/output/powerpc64le-linux-gnu/bitcoin-e40779a4fee0-powerpc64le-linux-gnu-debug.tar.gz
30b1c76daea40d40d487f4bb328d6191bc8c63b24822bde55160e6480b3a20d8  guix-build-e40779a4fee0/output/powerpc64le-linux-gnu/bitcoin-e40779a4fee0-powerpc64le-linux-gnu.tar.gz
50aafa855382d39703d9c67b17d3e7e6c2efff80d56638d4fd9e3cafce98cb67  guix-build-e40779a4fee0/output/riscv64-linux-gnu/SHA256SUMS.part
e71f5df0132edfe1b01b1a58040ba724976c292c362faba594476d6418545d8c  guix-build-e40779a4fee0/output/riscv64-linux-gnu/bitcoin-e40779a4fee0-riscv64-linux-gnu-debug.tar.gz
55d86637c60efd705d19006c7811dd26a11cae499da722e914ab61b992274a90  guix-build-e40779a4fee0/output/riscv64-linux-gnu/bitcoin-e40779a4fee0-riscv64-linux-gnu.tar.gz
87ec8f81933aae0c79162ab4084c9212caa6a93851fda413834edd41fec4a8bd  guix-build-e40779a4fee0/output/x86_64-apple-darwin/SHA256SUMS.part
0f858d52b37e6a5bda6b8bbd31c34a97e131696ae0803a183a760b1a72575ea5  guix-build-e40779a4fee0/output/x86_64-apple-darwin/bitcoin-e40779a4fee0-x86_64-apple-darwin-unsigned.dmg
25261f6401940a301ddb9c423bdf2c6c0721a86ae10a9e923264da7411ebec6e  guix-build-e40779a4fee0/output/x86_64-apple-darwin/bitcoin-e40779a4fee0-x86_64-apple-darwin-unsigned.tar.gz
3bf028ca001a013d21216eaa773d68cb709fab697b1ae2525589d110ce1dc672  guix-build-e40779a4fee0/output/x86_64-apple-darwin/bitcoin-e40779a4fee0-x86_64-apple-darwin.tar.gz
e4eabbabf4c276684123b6240b92f8389906827ed1a5231153481b8b1ce6f3ea  guix-build-e40779a4fee0/output/x86_64-linux-gnu/SHA256SUMS.part
74f8cee7c92b3aa5b7ec9c4a7cc44eb140aa11ec269fa7c1cb0c905feaf1f47c  guix-build-e40779a4fee0/output/x86_64-linux-gnu/bitcoin-e40779a4fee0-x86_64-linux-gnu-debug.tar.gz
a16923f7262c1a2588d9fdac4a3aae08eb77f349a6e035ccbf76b4529c5b6f34  guix-build-e40779a4fee0/output/x86_64-linux-gnu/bitcoin-e40779a4fee0-x86_64-linux-gnu.tar.gz
b2b148cddf4d40611e488630c7999c0a8f1996bb3e7e3788546b0598a5f5e6d1  guix-build-e40779a4fee0/output/x86_64-w64-mingw32/SHA256SUMS.part
2014b814af6d0b2c1382d2fb6c935d5b75a9b026c3e70679f867661461b1526c  guix-build-e40779a4fee0/output/x86_64-w64-mingw32/bitcoin-e40779a4fee0-win64-debug.zip
66c8606e8232aeb94f1f1c1bed00e8b22181128f3b5dbe4c332f6c6518751c29  guix-build-e40779a4fee0/output/x86_64-w64-mingw32/bitcoin-e40779a4fee0-win64-setup-unsigned.exe
741050ab2a4851c1510e23f5ad8f709f2318b55f2b6aa6dfac3da5d96c2ef36b  guix-build-e40779a4fee0/output/x86_64-w64-mingw32/bitcoin-e40779a4fee0-win64-unsigned.tar.gz
4d1d951c73fabde7cd5443641e658c9478fa6e5050401058504e972abd6d856f  guix-build-e40779a4fee0/output/x86_64-w64-mingw32/bitcoin-e40779a4fee0-win64.zip
```

fanquake reviewed (approved):
ACK e40779a4fee03c6c455149bd8e9d1a7ccd991450
